SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -1020,59 +1020,20 @@ void USpatialNetDriver::SpatialProcessServerTravel(const FString& URL, bool bAbs
 #endif // WITH_SERVER_CODE
 }
 
-void USpatialNetDriver::BeginDestroy()
+void USpatialNetDriver::PostInitProperties()
 {
-	Super::BeginDestroy();
+	Super::PostInitProperties();
 
-	if (Connection != nullptr)
+	if (!HasAnyFlags(RF_ClassDefaultObject))
 	{
-		// Delete all load-balancing partition entities if we're translator authoritative.
-		if (VirtualWorkerTranslationManager != nullptr)
-		{
-			for (const auto& Partition : VirtualWorkerTranslationManager->GetAllPartitions())
-			{
-				Connection->SendDeleteEntityRequest(Partition.PartitionEntityId, SpatialGDK::RETRY_UNTIL_COMPLETE);
-			}
-		}
-
-		if (RoutingSystem)
-		{
-			RoutingSystem->Destroy(Connection);
-
-			Connection->Flush();
-			FPlatformProcess::Sleep(0.1f);
-		}
-
-		if (StrategySystem)
-		{
-			StrategySystem->Destroy(Connection);
-
-			Connection->Flush();
-			FPlatformProcess::Sleep(0.1f);
-		}
-
-		// Cleanup our corresponding worker entity if it exists.
-		if (WorkerEntityId != SpatialConstants::INVALID_ENTITY_ID)
-		{
-			Connection->SendDeleteEntityRequest(WorkerEntityId, SpatialGDK::RETRY_UNTIL_COMPLETE);
-
-			// Flush the connection and wait a moment to allow the message to propagate.
-			// TODO: UNR-3697 - This needs to be handled more correctly
-			Connection->Flush();
-			FPlatformProcess::Sleep(0.1f);
-		}
-
-		// Destroy the connection to disconnect from SpatialOS if we aren't meant to persist it.
-		if (!bPersistSpatialConnection)
-		{
-			if (UWorld* LocalWorld = GetWorld())
-			{
-				Cast<USpatialGameInstance>(LocalWorld->GetGameInstance())->DestroySpatialConnectionManager();
-			}
-			Connection = nullptr;
-		}
+		// GuidCache will be allocated as an FNetGUIDCache above. To avoid an engine code change, we re-do it with the Spatial equivalent.
+		GuidCache = MakeShareable(new FSpatialNetGUIDCache(this));
```

Comment:
This should work as well:
```suggestion
		GuidCache = MakeShared<FSpatialNetGUIDCache>(this);
```